### ISSUE TITLE
Bug report: Path Param encoding not in sync with Zuul Retry. 

### ISSUE THREAD

**Issue by puneet739** [spring-cloud/spring-cloud-netflix#3388]
Hi Team,
This is related to https://github.com/spring-cloud/spring-cloud-netflix/issues/2178. 

When we are trying to access PathParam encoding in combination with Zuul Retry Functionality its breaking the path param and try to decode while sending to client. 

For Example: 
when we send '/context/a%2Fb/c/d' --> /context/a/b/c/d
when we have 
zuul:
  decodeUrl: true
  retryable: true

The root cause for this is at RetryableRibbonLoadBalancingHttpClient. 
The path we are creating the request for retry
```
if (service != null) {
					//Reconstruct the request URI using the host and port set in the retry context
					newRequest = newRequest.withNewUri(UriComponentsBuilder.newInstance().host(service.getHost())
							.scheme(service.getUri().getScheme()).userInfo(newRequest.getURI().getUserInfo())
							.port(service.getPort()).path(newRequest.getURI().getPath())
							.query(newRequest.getURI().getQuery()).fragment(newRequest.getURI().getFragment())
							.build().encode().toUri());
					
					if (ribbonLoadBalancerContext == null) {
						LOGGER.error("RibbonLoadBalancerContext is null. Unable to update load balancer stats");
					} 
```

**Comment by puneet739** [comment]
The issue is encountered over Greenwitch Release.

**Comment by puneet739** [comment]
@ryanjbaxter  you have added the release at 2.0.3, we have faced this issue over 2.1.2 Can you please suggest if this will be part of 2.1.X release as well?

**Comment by ryanjbaxter** [comment]
That is because the problem is also present in the 2.0.x branch.  I dont see how you could have seen this issue in 2.1.2 since that release doesn't exist.  Anyways, once the fix is in 2.0.x it will be merged forward into master (2.1.x).

**Comment by ryanjbaxter** [comment]
Closed via f55c95b
